pkg/network/server.go

```diff
@@ -238,11 +238,26 @@ func newServerFromConstructors(config ServerConfig, chain Ledger, stSync StateSy
 		s.BroadcastFactor = defaultBroadcastFactor
 	}
 
-	s.transport = newTransport(s)
+	if len(s.ServerConfig.Addresses) == 0 {
+		return nil, errors.New("no bind addresses configured")
+	}
+	var (
+		transports = make([]Transporter, len(s.ServerConfig.Addresses))
+		err        error
+	)
+	for i, addr := range s.ServerConfig.Addresses {
+		transports[i], err = newTransport(s, addr.Address)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create transporter: %w", err)
+		}
+	}
+	s.transports = transports
 	s.discovery = newDiscovery(
 		s.Seeds,
 		s.DialTimeout,
-		s.transport,
+		// Here we need to pick up a single transporter, it will be used to
+		// deal, and it doesn't matter which one.
```

Comment:
dial, it certainly can be improved to not do this in the future.

Reply:
We need to refactor transport then, create a separate issue?

Reply:
Transports were created to have different transport layer protocols (TCP, UDP, Websockets, etc), maybe they do have some meaning for dialers as well, but we only have IP addresses used for TCP now. Let it stay as is.